core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -426,7 +425,14 @@ private class InProcessServerStream implements ServerStream {
 
       InProcessServerStream(MethodDescriptor<?, ?> method, Metadata headers) {
         statsTraceCtx = StatsTraceContext.newServerContext(
-            serverStreamTracerFactories, method.getFullMethodName(), headers);
+          serverStreamTracerFactories, method.getFullMethodName(), headers);
+
+        syncContext = new SynchronizationContext(new Thread.UncaughtExceptionHandler() {
```

Comment:
Let's create this exception handler once in the transport, and reuse it for each of the streams.